src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -115,67 +115,28 @@ internal sealed class Arguments : TransformInputBase
             public bool OutputTokens;
 
             [Argument(ArgumentType.Multiple, HelpText = "A dictionary of whitelisted terms.", ShortName = "dict", NullName = "<None>", SortOrder = 10, Hide = true)]
-            public TermLoaderArguments Dictionary;
+            internal TermLoaderArguments Dictionary;
 
             [TGUI(Label = "Word Gram Extractor")]
-            [Argument(ArgumentType.Multiple, HelpText = "Ngram feature extractor to use for words (WordBag/WordHashBag).", ShortName = "wordExtractor", NullName = "<None>", SortOrder = 11)]
-            public INgramExtractorFactoryFactory WordFeatureExtractor = new NgramExtractorTransform.NgramExtractorArguments();
+            [Argument(ArgumentType.Multiple, Name = "WordFeatureExtractor", HelpText = "Ngram feature extractor to use for words (WordBag/WordHashBag).", ShortName = "wordExtractor", NullName = "<None>", SortOrder = 11)]
+            internal INgramExtractorFactoryFactory WordFeatureExtractorFactory = new NgramExtractorTransform.NgramExtractorArguments();
+
+            public WordBagEstimator.Options WordFeatureExtractor;
 
             [TGUI(Label = "Char Gram Extractor")]
-            [Argument(ArgumentType.Multiple, HelpText = "Ngram feature extractor to use for characters (WordBag/WordHashBag).", ShortName = "charExtractor", NullName = "<None>", SortOrder = 12)]
-            public INgramExtractorFactoryFactory CharFeatureExtractor = new NgramExtractorTransform.NgramExtractorArguments() { NgramLength = 3, AllLengths = false };
+            [Argument(ArgumentType.Multiple, Name = "CharFeatureExtractor", HelpText = "Ngram feature extractor to use for characters (WordBag/WordHashBag).", ShortName = "charExtractor", NullName = "<None>", SortOrder = 12)]
+            internal INgramExtractorFactoryFactory CharFeatureExtractorFactory = new NgramExtractorTransform.NgramExtractorArguments() { NgramLength = 3, AllLengths = false };
```

Comment:
>INgramExtractorFactoryFactory [](start = 21, length = 29)

FactoryFactory :), is it possible just have name INgramExtractorFactory

Reply:
Good point but we already have a class name `INgramExtractorFactory`
[https://github.com/dotnet/machinelearning/blob/ce0c917d1d8e44a5460f984d667107eecc4a9f78/src/Microsoft.ML.Transforms/Text/WordBagTransform.cs#L435](https://github.com/dotnet/machinelearning/blob/ce0c917d1d8e44a5460f984d667107eecc4a9f78/src/Microsoft.ML.Transforms/Text/WordBagTransform.cs#L435)
and there is relation between `INgramExtractorFactory` and `INgramExtractorFactoryFactory`.

---
In reply to: [264465575](https://github.com/dotnet/machinelearning/pull/2911#discussion_r264465575) [](ancestors = 264465575)